src/Bicep.Core.RegistryClient/BicepRegistryBlobClient.cs

```diff
@@ -0,0 +1,97 @@
+        private readonly ContainerRegistryBlobRestClient _blobRestClient;
+        private readonly string _repositoryName;
+
+        public BicepRegistryBlobClient(Uri endpoint, TokenCredential credential, string repositoryName) : this(endpoint, credential, repositoryName, new ContainerRegistryClientOptions())
+        {
+        }
+
+        public BicepRegistryBlobClient(Uri endpoint, TokenCredential credential, string repositoryName, ContainerRegistryClientOptions options)
+        {
+            Argument.AssertNotNull(endpoint, nameof(endpoint));
+            Argument.AssertNotNull(credential, nameof(credential));
+            Argument.AssertNotNull(options, nameof(options));
+
+            _endpoint = endpoint;
+            _registryName = endpoint.Host.Split('.')[0];
+            _repositoryName = repositoryName;
+            _clientDiagnostics = new ClientDiagnostics(options);
+
+            _acrAuthPipeline = HttpPipelineBuilder.Build(options);
+            _acrAuthClient = new AuthenticationRestClient(_clientDiagnostics, _acrAuthPipeline, endpoint.AbsoluteUri);
+
+            _pipeline = HttpPipelineBuilder.Build(options, new ContainerRegistryChallengeAuthenticationPolicy(credential, options.AuthenticationScope, _acrAuthClient));
+            _restClient = new ContainerRegistryRestClient(_clientDiagnostics, _pipeline, _endpoint.AbsoluteUri);
+            _blobRestClient = new ContainerRegistryBlobRestClient(_clientDiagnostics, _pipeline, _endpoint.AbsoluteUri);
+        }
+
+        public async Task<Response<UploadBlobResult>> UploadBlobAsync(Stream stream, CancellationToken cancellationToken = default)
+        {
+            string digest = DigestHelper.ComputeDigest(DigestHelper.AlgorithmIdentifierSha256, stream);
+
+            ResponseWithHeaders<ContainerRegistryBlobStartUploadHeaders> startUploadResult =
+                await _blobRestClient.StartUploadAsync(_repositoryName, cancellationToken).ConfigureAwait(false);
+
+            stream.Position = 0;
+            ResponseWithHeaders<ContainerRegistryBlobUploadChunkHeaders> uploadChunkResult =
+                await _blobRestClient.UploadChunkAsync(startUploadResult.Headers.Location, stream, cancellationToken).ConfigureAwait(false);
+
+            ResponseWithHeaders<ContainerRegistryBlobCompleteUploadHeaders> completeUploadResult =
+                await _blobRestClient.CompleteUploadAsync(digest, uploadChunkResult.Headers.Location, null, cancellationToken).ConfigureAwait(false);
+
+            return Response.FromValue(new UploadBlobResult(), completeUploadResult.GetRawResponse());
+        }
+
+        public async Task<Response<UploadManifestResult>> UploadManifestAsync(Stream stream, UploadManifestOptions options = default, CancellationToken cancellationToken = default)
+        {
+            options ??= new UploadManifestOptions();
+
+            string reference = options.Tag ?? DigestHelper.ComputeDigest(DigestHelper.AlgorithmIdentifierSha256, stream);
```

Comment:
>Tag [](http://example.com/codeflow?start=39&length=3)

@annelo-msft I added the tag as an optional property so we can associate it with the uploaded manifest.